contrib/babelfishpg_tsql/sql/sys_views.sql

```diff
@@ -748,32 +748,86 @@ and has_sequence_privilege(pg_get_serial_sequence(quote_ident(ext.nspname)||'.'|
-select
-  i.indrelid as object_id
-  , c.relname as name
-  , case when i.indisclustered then 1 else 2 end as type
-  , case when i.indisclustered then 'CLUSTERED'::varchar(60) else 'NONCLUSTERED'::varchar(60) end as type_desc
-  , case when i.indisunique then 1 else 0 end as is_unique
-  , c.reltablespace as data_space_id
-  , 0 as ignore_dup_key
-  , case when i.indisprimary then 1 else 0 end as is_primary_key
-  , case when constr.oid is null then 0 else 1 end as is_unique_constraint
-  , 0 as fill_factor
-  , case when i.indpred is null then 0 else 1 end as is_padded
-  , case when i.indisready then 0 else 1 end is_disabled
-  , 0 as is_hypothetical
-  , 1 as allow_row_locks
-  , 1 as allow_page_locks
-  , 0 as has_filter
-  , null::varchar as filter_definition
-  , 0 as auto_created
-  , c.oid as index_id
-from pg_class c
-inner join sys.schemas sch on c.relnamespace = sch.schema_id
-inner join pg_index i on i.indexrelid = c.oid
-left join pg_constraint constr on constr.conindid = c.oid
-where c.relkind = 'i' and i.indislive
-and has_schema_privilege(sch.schema_id, 'USAGE');
+select 
+  CAST(object_id as int)
+  , CAST(name as sys.sysname)
+  , CAST(type as sys.tinyint)
+  , CAST(type_desc as sys.nvarchar(60))
+  , CAST(is_unique as sys.bit)
+  , CAST(data_space_id as int)
+  , CAST(ignore_dup_key as sys.bit)
+  , CAST(is_primary_key as sys.bit)
+  , CAST(is_unique_constraint as sys.bit)
+  , CAST(fill_factor as sys.tinyint)
+  , CAST(is_padded as sys.bit)
+  , CAST(is_disabled as sys.bit)
+  , CAST(is_hypothetical as sys.bit)
+  , CAST(allow_row_locks as sys.bit)
+  , CAST(allow_page_locks as sys.bit)
+  , CAST(has_filter as sys.bit)
+  , CAST(filter_definition as sys.nvarchar)
+  , CAST(auto_created as sys.bit)
+  , CAST(index_id as int)
+from 
+(
+  -- Get all indexes from all system and user tables
+  select
+    i.indrelid as object_id
+    , c.relname as name
+    , case when i.indisclustered then 1 else 2 end as type
+    , case when i.indisclustered then 'CLUSTERED' else 'NONCLUSTERED' end as type_desc
+    , case when i.indisunique then 1 else 0 end as is_unique
+    , c.reltablespace as data_space_id
+    , 0 as ignore_dup_key
+    , case when i.indisprimary then 1 else 0 end as is_primary_key
+    , case when constr.oid is null then 0 else 1 end as is_unique_constraint
+    , 0 as fill_factor
+    , case when i.indpred is null then 0 else 1 end as is_padded
+    , case when i.indisready then 0 else 1 end as is_disabled
+    , 0 as is_hypothetical
+    , 1 as allow_row_locks
+    , 1 as allow_page_locks
+    , 0 as has_filter
+    , null as filter_definition
+    , 0 as auto_created
+    , case when i.indisclustered then 1 else c.oid end as index_id
+  from pg_class c
+  inner join pg_index i on i.indexrelid = c.oid
+  left join pg_constraint constr on constr.conindid = c.oid
+  where c.relkind = 'i' and i.indislive
+  and (c.relnamespace in (select schema_id from sys.schemas) or c.relnamespace::regnamespace::text = 'sys')
```

Comment:
We shouldn't be doing that regnamespace casting dance unless we have a really good reason to do so (and if so it needs to be documented)

Reply:
I have created an issue to address this in the future here: https://github.com/babelfish-for-postgresql/babelfish_extensions/issues/190

This will take some time to change & properly test but I want to unblock other developers who require some of the additions in this PR ASAP. 

The view still works, and from the customer's perspective they won't see a difference if casting is in the implementation or not, but I do agree a change is in order and this casting stuff isn't great.

Reply:
Agreed, but please note the comment I just added to #190

Reply:
Ack